src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,215 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private MyLinkedList.Node<T> first;
+    private MyLinkedList.Node<T> last;
+
+    MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = new Node<>(null, value, null);
+            last = first;
+            size++;
+        } else {
+            Node node = new Node<>(last, value, null);
+            last.setNext(node);
+            last = node;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        validateAddIndex(index);
+        Node node = first;
+        Node newNode = new Node<>(value);
+        if (node == null) {
+            first = newNode;
+            last = newNode;
+            size++;
+            return;
+        }
+        for (int i = 0; i <= size; i++) {
```

Comment:
do not use loop here, create getNodeByIndex() method and reuse it here and in remove method

Reply:
done